src/testdrive/src/action/kafka/await_ingestion.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::time::Duration;
+
+use anyhow::bail;
+use mz_ore::collections::CollectionExt;
+use mz_ore::retry::Retry;
+use mz_pgrepr::{Numeric, UInt8};
+use mz_repr::adt::range::Range;
+use mz_storage_types::sources::MzOffset;
+use mz_timely_util::antichain::AntichainExt;
+use mz_timely_util::order::Partitioned;
+use rdkafka::admin::AdminClient;
+use rdkafka::client::DefaultClientContext;
+use timely::progress::Antichain;
+use timely::PartialOrder;
+
+use crate::action::{ControlFlow, State};
+use crate::parser::BuiltinCommand;
+
+pub async fn run_await_ingestion(
+    mut cmd: BuiltinCommand,
+    state: &mut State,
+) -> Result<ControlFlow, anyhow::Error> {
+    let topic = cmd.args.string("topic")?;
+    let source = cmd.args.opt_string("source");
+    let progress_subsource = cmd.args.opt_string("progress-subsource");
+    cmd.args.done()?;
+
+    let progress_subsource = match (source, progress_subsource) {
+        (None, Some(progress)) => progress,
+        (Some(source), None) => format!("{source}_progress"),
+        (None, None) => bail!("Missing argument. Specify one of `source` or `progress-subsource`"),
+        (Some(_), Some(_)) => {
+            bail!("Conflicting arguments. Specify one of `source` or `progress-subsource`")
+        }
+    };
+
+    let topic = format!("testdrive-{}-{}", topic, state.seed);
+    let config = state.kafka_config.clone();
+
+    // First let's get the upper frontier of the kafka topic
+    let current_upper = mz_ore::task::spawn_blocking(
+        || "kakfa_watermark".to_string(),
+        move || {
+            let client: AdminClient<DefaultClientContext> = config.create()?;
+            let meta = client
+                .inner()
+                .fetch_metadata(Some(&topic), Duration::from_secs(10))?;
```

Comment:
This needs to use the global testdrive timeout instead of the hard-coded 10 sec.